src/richie/apps/search/index_manager.py

```diff
@@ -112,15 +113,34 @@ def regenerate_indexes(logger):
     # NB: we *must* do this before the update_aliases call so we don't immediately prune
     # version n-1 of all our indexes
     useless_indexes = [
-        index for index, details in existing_indexes.items() if "aliases" not in details
+        index for index, details in existing_indexes.items() if details["aliases"] == {}
     ]
 
     # Replace the old indexes with the new ones in 1 atomic operation to avoid outage
-    indices_client.update_aliases(
-        dict(actions=actions_to_create_aliases + actions_to_delete_aliases)
-    )
+    def perform_aliases_update():
+        try:
+            indices_client.update_aliases(
+                dict(actions=actions_to_create_aliases + actions_to_delete_aliases)
+            )
+        except RequestError as exception:
+            # This operation can fail if an index exists with the same name as an alias we're
```

Comment:
I thought of a case which might be problematic. 

Let's say you have a Richie site running with the default index name and  you add a new site to the same ES cluster. If you don't customize the index name, the aliases will be in conflict and this code will attempt to drop the index of the existing site without warning... Not sure what happens in this case as it is an alias and not an index but we must check and maybe add a test? I might also be missing something...

Reply:
Not sure how we would have a conflict here, as we'd have eg `richie_courses` from the first default install and then `prefixed_richie_courses` from the other one

Reply:
(Note that this PR does not include the changes related to index name variabilization)

Reply:
sorry I forgot to mention that in my scenario, you would start the second project with the default index name as well (because you didn't know about this setting). You first index would at risk of this kind of negligence.

Reply:
Yes, it would be broken to try and run two projects without prefixing. However, it is currently broken and this PR does not change that.

Maybe there's a way to mitigate this, IMO it should then be addressed in the next PR that deals with variabilization of index names?